nesis/api/core/document_loaders/sharepoint.py

```diff
@@ -23,158 +27,278 @@
 _sharepoint_context = None
 
 
-def fetch_documents(**kwargs) -> None:
+def fetch_documents(
+    connection: Dict[str, str],
+    rag_endpoint: str,
+    http_client: http.HttpClient,
+    metadata: Dict[str, Any],
+    cache_client: memcache.Client,
+) -> None:
+    global _sharepoint_context
```

Comment:
@akizito I think it'd be best to have this as a local variable rather than global and then pass it down to the `_sync_sharepoint_documents` and `_unsync_sharepoint_documents` functions. This way, this `fetch_documents` becomes thread safe - if in future we want to use it in a multi-threaded manner